code/modules/mob/living/simple_animal/friendly/diona.dm

```diff
@@ -92,6 +92,20 @@
 /mob/living/simple_animal/diona/UnarmedAttack(var/atom/A)
 	if(isdiona(A) && (src in A.contents)) //can't attack your gestalt
 		visible_message("[src] wiggles around a bit.")
+	// Nymphs can consume grown food as well as weeds.
+	else if(istype(A, /obj/item/reagent_containers/food/snacks/grown))
+		if(nutrition >= nutrition_need) // Prevents griefing by overeating plant items without evolving.
+			to_chat(src, "<span class='warning'>You're too full to consume this! Perhaps it's time to grow bigger...</span>")
+		else
+			if(do_after_once(src, 20, target = A))
+				visible_message("[src] ravenously consumes [A].")
+				playsound(loc, 'sound/misc/demon_consume.ogg', 30, 0, frequency = 1.5)
```

Comment:
I always knew nymphs were demonic.

Reply:
It's true! Also no, they still have to take blood, it just lets them consume plant matter other than weeds to gain nutrients.

Reply:
Though I do take issue with using this sound; it should be the same generic eat sound everyone else has.